vstools/functions/utils.py

```diff
@@ -54,6 +54,49 @@ class DitherType(CustomStrEnum):
     ERROR_DIFFUSION = 'error_diffusion'
     """Floyd-Steinberg error diffusion."""
 
+    ERROR_DIFFUSION_FMTC = 'error_diffusion_fmtc'
+    """Floyd-Steinberg error diffusion, modified for serpentine scan (avoids worm artefacts)."""
+
+    SIERRA_LITE = 'sierra_lite'
+    """Another type of error diffusion with more focus on speed."""
+
+    STUCKI = 'stucki'
+    """Another error diffusion kernel that preserves delicate edges better but distorts gradients."""
+
+    ATKINSON = 'atkinson'
+    """Another error diffusion kernel that generates distinct patterns but keeps clean the flat areas (noise modulation)."""
+
+    OSTROMOUKHOV = 'ostromoukhov'
+    """Another error diffusion kernel. Slow, available only for integer input at the moment. Avoids usual F-S artefacts."""
+
+    VOID = 'void'
+    """A way to generate blue-noise dither and has a much better visual aspect than ordered dithering."""
+
+    QUASIRANDOM = 'quasirandom'
+    """Dither using quasirandom sequences. Good intermediated between Void and cluster and error diffusion algorithms."""
+
+    def apply(self, clip: vs.VideoNode, format: vs.VideoFormat, range_in: ColorRange, range_out: ColorRange, format_in: vs.VideoFormat) -> vs.VideoNode:
```

Comment:
Format overrides a python builtin, you should rename both to ftm_in and ftm_out, following the range naming scheme, and maybe reorder as fmt in,out range in,out.
Also do we really need format_in for checking the vs.FLOAT? Can't you just see from clip.format.sample_type